src/components/main.tsx

```diff
@@ -54,8 +71,9 @@ const Main = () => {
                         spellCheck={false}
                         autoComplete="off"
                         className="main-input"
-                        placeholder="Enter your Ethereum address"
+                        placeholder="Enter your Ethereum address or ENS Domain"
                         value={input}
+                        invalid={!validAddress && !!input}
```

Comment:
Nice that it also validates regular addresses! I like it.

I have a strong objection to one part of the current behavior though. From a UX perspective, I can't stand when password entries, form validations etc. tell me my input is wrong until it is right. 
Ideally, only validate the form on submission please so that I am not getting an invalid input warning the whole time I am typing in a an ens domain.